### PR TITLE
Add a note on autocomplete

### LINKED ISSUE #314: Add note on autofill to a11y checklist
On the [secure markup page](https://github.com/springernature/frontend-playbook/blob/master/security/secure-markup.md) there's a note on autofill that goes over the reasons that we don't disable autofill for security reasons, and why we shouldn't autofill fields that are visually hidden. 

There are also accessibility reasons for not disabling the `autocomplete` functionality for any fields containing personal information - it can make it more difficult to use password managers or form fillers, which is a major negative for anyone whose memory isn't too good, or who struggles to type quickly into input fields (e.g. switch device users). 

We should add a note about that to the a11y checklist, and link back to the note on the secure markup page too.

### PR DESCRIPTION
**Pull request by hollsk**
resolves #314 

* Creates a new "forms" section that contains checklist items about forms
* Adds guidance about autocomplete/autofill from an accessibility perspective

### CODE REVIEW

**Comment by jon-stevens** [comment]
image

File: accessibility/accessibility-checklist.md [hunk 1]
```diff
@@ -228,6 +226,14 @@ Resources:
 * [Article: Opening links in a new window](https://www.nomensa.com/blog/2011/opening-links-new-window)
 
 
+## Forms
+
+### Labels and context
+
+Associate all form controls with appropriate labels.
```
Comment (alexkilgour) [inline accessibility/accessibility-checklist.md]: could be a 2 item list?

**Comment by alexkilgour** [comment]
Looks good and makes sense. Do we want to expand on any other aspects of forms? for example, at the moment we are looking at issues such as (1) Should we use multi-stage forms (wizard style, over multiple pages), or (2) Should we use interactive forms (form fields update via JS based on previous answers)

File: accessibility/accessibility-checklist.md [hunk 2]
```diff
@@ -240,6 +246,25 @@ Resources:
...
+
+Don't indiscriminately disable autocomplete for input fields. The default autocomplete and autofill behaviours are useful for many users with and without disabilities. 
+
+For example, autocomplete helps users with motor disabilities by reducing the amount of typing they need to do in input fields. Autofill can help users with cognitive disabilities by automatically filling in common information like addresses or telephone numbers. 
+
+Don't disable autocomplete on sensitive fields like username and password fields. Any security benefits of doing this is negligible compared to the benefits of users' in-browser password managers, and has a direct negative impact on accessibility. Read the note on autofill in the [secure markup guide](https://github.com/springernature/frontend-playbook/blob/master/security/secure-markup.md#a-note-on-autofill) for more on that.
```
Comment (jpw) [inline accessibility/accessibility-checklist.md]: <3

**Comment by hollsk** [comment]
@alexkilgour 
>Looks good and makes sense. Do we want to expand on any other aspects of forms? for example, at the moment we are looking at issues such as (1) Should we use multi-stage forms (wizard style, over multiple pages), or (2) Should we use interactive forms (form fields update via JS based on previous answers)

Ooh, maybe. Do we have a firm conclusion? Without being part of those discussions, it looks to me like that has some nuance involved which might justify a fuller treatment than just one or two lines in the checklist?

**Comment by alexkilgour** [comment]
> Ooh, maybe. Do we have a firm conclusion? Without being part of those discussions, it looks to me like that has some nuance involved which might justify a fuller treatment than just one or two lines in the checklist?

No firm conclusions, and yes in hindsight I think it is out of scope for this and needs more thought. But it is a subject that I think would benefit from some documented approaches as I think this would definitely help teams.

**Pull request merged**